addons/packages/vsphere-csi/2.4.1/bundle/config/overlays/update-csi-driver.yaml

```diff
@@ -0,0 +1,150 @@
+#@ load("/values.star", "values")
+#@ load("@ytt:overlay", "overlay")
+#@ load("@ytt:data", "data")
+#@ load("@ytt:yaml", "yaml")
+#@ load("@ytt:sha256", "sha256")
+
+#@overlay/match by=overlay.subset({"kind": "Deployment", "metadata": {"name": "vsphere-csi-controller"}})
+---
+metadata:
+  #@overlay/match missing_ok=True
+  annotations:
+    kapp.k14s.io/disable-default-label-scoping-rules: ""
+    kapp.k14s.io/update-strategy: fallback-on-replace
+    vsphere-csi/data-values-hash: #@ sha256.sum(yaml.encode(values))
+  namespace: #@ values.vsphereCSI.namespace
+
+spec:
+  #@overlay/match
+  replicas: #@ values.vsphereCSI.deployment_replicas
+  template:
+    spec:
+      containers:
+        #@overlay/match by=overlay.subset({"name": "csi-attacher"})
+        - args:
+            #@overlay/match by=lambda index, left, right: "--timeout=" in left, expects=1
+            - #@ "--timeout=" + values.vsphereCSI.attachTimeout
+              #@ if values.vsphereCSI.http_proxy != "" :
+            #@overlay/match by=overlay.subset({"name": "vsphere-csi-controller"})
+        - env:
+            #@overlay/append
+            - name: "HTTP_PROXY"
+              value: #@ values.vsphereCSI.http_proxy
+            - name: "HTTPS_PROXY"
+              value: #@ values.vsphereCSI.https_proxy
+            - name: "NO_PROXY"
+              value: #@ values.vsphereCSI.no_proxy
+        #@ end
+        #@ if values.vsphereCSI.http_proxy != "" :
+        #@overlay/match by=overlay.subset({"name": "vsphere-syncer"})
+        - env:
+            #@overlay/append
+            - name: "HTTP_PROXY"
+              value: #@ values.vsphereCSI.http_proxy
+            - name: "HTTPS_PROXY"
+              value: #@ values.vsphereCSI.https_proxy
+            - name: "NO_PROXY"
+              value: #@ values.vsphereCSI.no_proxy
+        #@ end
+        #@overlay/match by=overlay.subset({"name": "csi-provisioner"})
+        - args:
+            #@overlay/match by=lambda index, left, right: "--timeout=" in left, expects=1
+            - #@ "--timeout=" + values.vsphereCSI.provisionTimeout
+              #@ if values.vsphereCSI.region or values.vsphereCSI.zone:
```

Comment:
Should we use `useTopologyFeature` for adding these 2 arguments? Also indentation for if/end is not aligned.

Reply:
Does older case (region and zone are set, topologyFeature is false) require these flags?

Reply:
IIUC, the enablement of `--feature-gates=Topology=true` is based on whether `values.vsphereCSI.region` or `values.vsphereCSI.zone` is passed in. `useTopologyFeature` has a confusing name, it actually tells whether to use `topology-categories` or `region, zone` in csi configs.

Reply:
Fixed the indentation.
Updated the flag name to `useTopologyCategories`